src/main/java/io/confluent/kafkarest/SimpleConsumerPool.java

```diff
@@ -35,7 +36,7 @@
   private final Time time;
 
   private final SimpleConsumerFactory simpleConsumerFactory;
-  private final Map<String, SimpleConsumer> simpleConsumers;
+  private final Map<String, Consumer<byte[], byte[]>> simpleConsumers;
```

Comment:
Hmm, I'm honestly not sure why this was previously implemented this way with a separate map and list of IDs. Think we could just make simpleConsumers just a Queue and get rid of available consumers? Although with the suggestion about caching consumers by topic partition for more effective reuse, we might need to further adjust this.